feat: 5201 - change currency with country when relevant

monsieurtanuki opened the pull request:
### What
We change the currency with the best possible matching value for a given country
- at onboarding time for new users, silently
- at app init time for old users that of course do not have a currency yet (recent field), silently
- when the user explicitly changes the country, via a "confirm?" dialog

### Screenshot
![Screenshot_1715344369](https://github.com/openfoodfacts/smooth-app/assets/11576431/07200de4-ff71-4e66-90dc-edc9a994e1ef)

### Fixes bug(s)
- Closes: #5201

### Impacted files
* `app_en.arb`: message for a "change currency too?" dialog
* `country_selector.dart`: new method to change the currency at the same time as the country if relevant
* `product_query.dart`: init currency if null
* `pubspec.lock`: wtf
* `pubspec.yaml`: upgraded `openfoodfacts` to `3.8.0`
* `user_preferences_country_selector.dart`: asking to change the currency at the same time as the country only if confirmed or relevant
* `welcome_page.dart`: explicitly asking to change the currency at the same time as the country

monsieurtanuki committed:
feat: 5201 - change currency with country when relevant

Impacted files:
* `app_en.arb`: message for a "change currency too?" dialog
* `country_selector.dart`: new method to change the currency at the same time as the country if relevant
* `product_query.dart`: init currency if null
* `pubspec.lock`: wtf
* `pubspec.yaml`: upgraded `openfoodfacts` to `3.8.0`
* `user_preferences_country_selector.dart`: asking to change the currency at the same time as the country only if confirmed or relevant
* `welcome_page.dart`: explicitly asking to change the currency at the same time as the country

monsieurtanuki committed:
refactoring

teolemon commented:
Can we split into 2 sentences/strings, and avoid ellisions (You've) ?
Can we avoid adding things to the onboarding until we have a compelling value proposition ? (And even so, I'd rather move to small mono-theme, in context onboardings)

monsieurtanuki commented:
> Can we split into 2 sentences/strings, and avoid ellisions (You've) ?

I can split into 2 sentences.
I'm not convinced by your idea of removing contractions - as explained here (https://m.youtube.com/watch?v=rNcS0S__WlQ), but as long as I split into 2 sentences you'll be able - I mean you will be able - to have it your way.

> Can we avoid adding things to the onboarding until we have a compelling value proposition ? (And even so, I'd rather move to small mono-theme, in context onboardings)

I haven't really added anything to the onboarding. It's just that I doesn't seem relevant to display `ADP` (aka Andorran peseta) as the selected currency for any user that hasn't set the currency yet.
Or maybe I misinterpreted your comment.

monsieurtanuki committed:
minor changes about labels

monsieurtanuki commented:
@teolemon Done.

monsieurtanuki committed:
Merge branch 'develop' into feat/5201

monsieurtanuki commented:
Thank you @teolemon for your review!
